winlogbeat/sys/wineventlog/wineventlog_windows.go

```diff
@@ -419,6 +420,46 @@ func FormatEventString(
 	return common.UTF16ToUTF8Bytes(buffer[:bufferUsed], out)
 }
 
+// Publishers returns a sort list of event publishers on the local computer.
+func Publishers() ([]string, error) {
+	publisherEnumerator, err := _EvtOpenPublisherEnum(NilHandle, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed in EvtOpenPublisherEnum: %w", err)
+	}
+	defer Close(publisherEnumerator)
+
+	var (
+		publishers []string
+		bufferUsed uint32
+		buffer     = make([]byte, 1024)
+	)
+
+loop:
+	for {
+		if err = _EvtNextPublisherId(publisherEnumerator, uint32(len(buffer)), &buffer[0], &bufferUsed); err != nil {
```

Comment:
This can potentially overflow the buffer argument, as the bufferSize parameter is count of characters, not bytes.
```suggestion
		if err = _EvtNextPublisherId(publisherEnumerator, uint32(len(buffer)/2), &buffer[0], &bufferUsed); err != nil {
```

Alternatively it might be simpler to change the function signature to receive `[]uint16`

Reply:
I switched it over to []uint16.